API/chat/forms.py

```diff
@@ -32,23 +32,12 @@ def save(self):
 
 
 class MessagePatchForm(forms.Form):
-    id = forms.IntegerField()
     text = forms.CharField(widget=forms.Textarea)
     datetime_sent = forms.IntegerField()
     typing = forms.BooleanField(required=False)
 
-    def is_valid(self):
-        if not super(MessagePatchForm, self).is_valid():
-            return False
-
-        message = Message.objects.get(pk=self.cleaned_data['id'])
-        if not message.typing:
-            return False
-
-        return True
-
     def save(self):
-        message = Message.objects.get(pk=self.cleaned_data['id'])
+        message = self.message
```

Comment:
Indeed, best practice is to define attributes at the class level or in __init__().
I personally don't like this design very much: There is no interface on the Form object that informs the user that at some point `message` needs to be assigned for save to work.
If it was up to me I would just have the message object as an argument on the appropriate function that needs it (save?) OR pass it to the form during initialisation. Both these solutions provide a clean explicit interface.